### ISSUE TITLE
Fix all EPUB validation failures/warnings

### ISSUE THREAD

**Issue by IntermittentlyRupert** [IntermittentlyRupert/hpnofp-ebook#7]

**Comment by ThrawnCA** [comment]
Do the `h3` elements in the table of contents actually achieve anything? If so, would they still work if wrapped in `li` tags?

**Comment by ThrawnCA** [comment]
As an aside to this, I want to refactor the build a bit, so you can choose a build target, eg `build.sh check` to run epubcheck, or `build.sh pdf` to generate PDF but not MOBI.

**Comment by IntermittentlyRupert** [comment]
> Do the `h3` elements in the table of contents actually achieve anything?

I've used them for the "**Year [n]**" section headings.

> If so, would they still work if wrapped in `li` tags?

Possibly? Each `li` might have to be a nav item, though... I forget the details, but I tried a few different options when I was setting it up originally. Because `TOC.xhtml` is registered as the table of contents in `book.opf` it has to satisfy certain restrictions - I remember that I originally wanted to have an `ol` for each year and it needed to have exactly one `ol` or `ul` element. There were also some issues with how the Kindle handled a couple of the options I tried, even though ePubCheck had no problem with them. 

It might actually be worth having two TOCs - one that's pretty-formatted at the front of the book (as a normal, non-semantic item), and one that's more functional at the back of the book (or maybe not in the book flow at all, if that's an option).
